src/libs/actions/IOU.js

```diff
@@ -752,6 +752,30 @@ function editDistanceMoneyRequest(transactionID, transactionThreadReportID, tran
         },
     });
 
+    // Update recently used categories if the category is changed
+    if (_.has(transactionChanges, 'category')) {
+        const optimisticPolicyRecentlyUsedCategories = Policy.buildOptimisticPolicyRecentlyUsedCategories(iouReport.policyID, transactionChanges.category);
+        if (!_.isEmpty(optimisticPolicyRecentlyUsedCategories)) {
+            optimisticData.push({
+                onyxMethod: Onyx.METHOD.SET,
+                key: `${ONYXKEYS.COLLECTION.POLICY_RECENTLY_USED_CATEGORIES}${iouReport.policyID}`,
+                value: optimisticPolicyRecentlyUsedCategories,
+            });
+        }
+    }
+
+    // Update recently used categories if the tag is changed
+    if (_.has(transactionChanges, 'tag')) {
+        const optimisticPolicyRecentlyUsedTags = Policy.buildOptimisticPolicyRecentlyUsedTags(iouReport.policyID, transactionChanges.tag);
+        if (!_.isEmpty(optimisticPolicyRecentlyUsedTags)) {
+            optimisticData.push({
+                onyxMethod: Onyx.METHOD.MERGE,
```

Comment:
@hoangzinh  I agree with you at this point. But currently we use MERGE method for tag in other places
https://github.com/Expensify/App/blob/18d4b9b661e77ff1d3e5d0c335df5c838c405345/src/libs/actions/IOU.js#L187

https://github.com/Expensify/App/blob/18d4b9b661e77ff1d3e5d0c335df5c838c405345/src/libs/actions/IOU.js#L1875

Reply:
@hoangzinh Do you mean that we should update to use SET in all places ?

Reply:
@hoangzinh Updated

Reply:
i had this exact question! 

The only thing though is that I think these all either need to be `SET` or all need to `MERGE` per
> For this reason, it is usually preferable to use one or the other, but not both.

And even more fun once you look at `buildOptimisticPolicyRecentlyUsedCategories` returns an `array` and `buildOptimisticPolicyRecentlyUsedTags` returns an `object`

So I think for calls to set the recently used categories we should use `SET` everywhere as noted
> You should avoid arrays as much as possible. They do not work well with merge() because it can't update a single element in an array, it must always set the entire array each time.

And for recently used tags we should use `MERGE` everywhere since those are objects.